diff -ru libspatialite-5.1.0-orig/makefile64.vc libspatialite-5.1.0/makefile64.vc
--- libspatialite-5.1.0-orig/makefile64.vc	2023-11-06 08:42:52.257245934 +1000
+++ libspatialite-5.1.0/makefile64.vc	2023-11-06 08:43:10.165569026 +1000
@@ -3,6 +3,7 @@
 # NMAKE Makefile to build libspatialite on Windows
 #
 !INCLUDE nmake64.opt
+INSTDIR=$(LIBRARY_PREFIX)
 
 LIBOBJ = src\gaiaaux\gg_sqlaux.obj src\gaiaaux\gg_utf8.obj \
 	src\gaiaexif\gaia_exif.obj src\gaiageo\gg_advanced.obj \
@@ -96,7 +97,7 @@
 SPATIALITE_DLL = spatialite$(VERSION).dll
 
 CFLAGS = /nologo -I.\src\headers -I.\src\topology \
-	-I. -IC:\OSGeo4W64\include -IC:\OSGeo4W64\include\libxml2 $(OPTFLAGS)
+	-I. -I$(LIBRARY_INC) -I$(LIBRARY_INC)\libxml2 $(OPTFLAGS)
 
 default:	all
 
@@ -112,10 +113,11 @@
 spatialite_i.lib:     $(LIBOBJ)
 	link /dll /out:$(SPATIALITE_DLL) \
 		/implib:spatialite_i.lib $(LIBOBJ) \
-		C:\OSGeo4W64\lib\proj_i.lib C:\OSGeo4W64\lib\geos_c.lib \
-		C:\OSGeo4w64\lib\freexl_i.lib C:\OSGeo4w64\lib\iconv.lib \
-		C:\OSGeo4W64\lib\sqlite3_i.lib C:\OSGeo4W64\lib\zlib.lib \
-		C:\OSGeo4W64\lib\libxml2.lib C:\OSGeo4W64\lib\librttopo.lib
+		$(LIBRARY_LIB)\proj.lib $(LIBRARY_LIB)\geos_c.lib \
+		$(LIBRARY_LIB)\freexl_i.lib $(LIBRARY_LIB)\iconv.lib \
+		$(LIBRARY_LIB)\sqlite3.lib $(LIBRARY_LIB)\zlib.lib \
+		$(LIBRARY_LIB)\libxml2.lib $(LIBRARY_LIB)\librttopo.lib \
+		$(LIBRARY_LIB)\charset.lib
 	if exist $(SPATIALITE_DLL).manifest mt -manifest \
 		$(SPATIALITE_DLL).manifest -outputresource:$(SPATIALITE_DLL);2
 		
diff -ru libspatialite-5.1.0-orig/makefile_mod64.vc libspatialite-5.1.0/makefile_mod64.vc
--- libspatialite-5.1.0-orig/makefile_mod64.vc	2023-11-06 08:42:52.257245934 +1000
+++ libspatialite-5.1.0/makefile_mod64.vc	2023-11-06 08:43:10.169569100 +1000
@@ -3,6 +3,7 @@
 # NMAKE Makefile to build libspatialite on Windows
 #
 !INCLUDE nmake_mod64.opt
+INSTDIR=$(LIBRARY_PREFIX)
 
 LIBOBJ = src\gaiaaux\gg_sqlaux.obj src\gaiaaux\gg_utf8.obj \
 	src\gaiaexif\gaia_exif.obj src\gaiageo\gg_advanced.obj \
@@ -96,7 +97,7 @@
 MOD_SPATIALITE_DLL = mod_spatialite$(VERSION).dll
 
 CFLAGS = /nologo -I.\src\headers -I.\src\topology \
-	-I. -IC:\OSGeo4W64\include -IC:\OSGeo4W64\include\libxml2 $(OPTFLAGS)
+	-I. -I$(LIBRARY_INC) -I$(LIBRARY_INC)\libxml2 $(OPTFLAGS)
 
 default:	all
 
@@ -112,10 +113,11 @@
 mod_spatialite_i.lib:     $(LIBOBJ)
 	link /dll /out:$(MOD_SPATIALITE_DLL) \
 		/implib:mod_spatialite_i.lib $(LIBOBJ) \
-		C:\OSGeo4W64\lib\proj_i.lib C:\OSGeo4W64\lib\geos_c.lib \
-		C:\OSGeo4w64\lib\freexl_i.lib C:\OSGeo4w64\lib\iconv.lib \
-		C:\OSGeo4W64\lib\sqlite3_i.lib C:\OSGeo4W64\lib\zlib.lib \
-		C:\OSGeo4W64\lib\libxml2.lib C:\OSGeo4W64\lib\librttopo.lib
+		$(LIBRARY_LIB)\proj.lib $(LIBRARY_LIB)\geos_c.lib \
+		$(LIBRARY_LIB)\freexl_i.lib $(LIBRARY_LIB)\iconv.lib \
+		$(LIBRARY_LIB)\sqlite3.lib $(LIBRARY_LIB)\zlib.lib \
+		$(LIBRARY_LIB)\libxml2.lib $(LIBRARY_LIB)\librttopo.lib \
+		$(LIBRARY_LIB)\charset.lib
 	if exist $(MOD_SPATIALITE_DLL).manifest mt -manifest \
 		$(MOD_SPATIALITE_DLL).manifest -outputresource:$(MOD_SPATIALITE_DLL);2
 		
diff -ru libspatialite-5.1.0-orig/makefile_mod.vc libspatialite-5.1.0/makefile_mod.vc
--- libspatialite-5.1.0-orig/makefile_mod.vc	2023-11-06 08:42:52.257245934 +1000
+++ libspatialite-5.1.0/makefile_mod.vc	2023-11-06 08:43:10.169569100 +1000
@@ -3,6 +3,7 @@
 # NMAKE Makefile to build libspatialite on Windows
 #
 !INCLUDE nmake_mod.opt
+INSTDIR=$(LIBRARY_PREFIX)
 
 LIBOBJ = src\gaiaaux\gg_sqlaux.obj src\gaiaaux\gg_utf8.obj \
 	src\gaiaexif\gaia_exif.obj src\gaiageo\gg_advanced.obj \
@@ -96,7 +97,7 @@
 MOD_SPATIALITE_DLL = mod_spatialite$(VERSION).dll
 
 CFLAGS = /nologo -I.\src\headers -I.\src\topology \
-	-I. -IC:\OSGeo4W\include $(OPTFLAGS)
+	-I. -I$(LIBRARY_INC) $(OPTFLAGS)
 
 default:	all
 
@@ -112,10 +113,11 @@
 mod_spatialite_i.lib:     $(LIBOBJ)
 	link /dll /out:$(MOD_SPATIALITE_DLL) \
 		/implib:mod_spatialite_i.lib $(LIBOBJ) \
-		C:\OSGeo4W\lib\proj_i.lib C:\OSGeo4W\lib\geos_c.lib \
-		C:\OSGeo4w\lib\freexl_i.lib C:\OSGeo4w\lib\iconv.lib \
-		C:\OSGeo4W\lib\sqlite3_i.lib C:\OSGeo4W\lib\zlib.lib \
-		C:\OSGeo4W\lib\libxml2.lib C:\OSGeo4W\lib\librttopo.lib
+		$(LIBRARY_LIB)\proj.lib $(LIBRARY_LIB)\geos_c.lib \
+		$(LIBRARY_LIB)\freexl_i.lib $(LIBRARY_LIB)\iconv.lib \
+		$(LIBRARY_LIB)\sqlite3.lib $(LIBRARY_LIB)\zlib.lib \
+		$(LIBRARY_LIB)\libxml2.lib $(LIBRARY_LIB)\librttopo.lib \
+		$(LIBRARY_LIB)\charset.lib
 	if exist $(MOD_SPATIALITE_DLL).manifest mt -manifest \
 		$(MOD_SPATIALITE_DLL).manifest -outputresource:$(MOD_SPATIALITE_DLL);2
 		
diff -ru libspatialite-5.1.0-orig/makefile.vc libspatialite-5.1.0/makefile.vc
--- libspatialite-5.1.0-orig/makefile.vc	2023-11-06 08:42:52.257245934 +1000
+++ libspatialite-5.1.0/makefile.vc	2023-11-06 08:43:10.169569100 +1000
@@ -3,6 +3,7 @@
 # NMAKE Makefile to build libspatialite on Windows
 #
 !INCLUDE nmake.opt
+INSTDIR=$(LIBRARY_PREFIX)
 
 LIBOBJ = src\gaiaaux\gg_sqlaux.obj src\gaiaaux\gg_utf8.obj \
 	src\gaiaexif\gaia_exif.obj src\gaiageo\gg_advanced.obj \
@@ -96,7 +97,7 @@
 SPATIALITE_DLL = spatialite$(VERSION).dll
 
 CFLAGS = /nologo -I.\src\headers -I.\src\topology \
-	-I. -IC:\OSGeo4W\include $(OPTFLAGS)
+	-I. -I$(LIBRARY_INC) $(OPTFLAGS)
 
 default:	all
 
@@ -112,10 +113,11 @@
 spatialite_i.lib:     $(LIBOBJ)
 	link /dll /out:$(SPATIALITE_DLL) \
 		/implib:spatialite_i.lib $(LIBOBJ) \
-		C:\OSGeo4W\lib\proj_i.lib C:\OSGeo4W\lib\geos_c.lib \
-		C:\OSGeo4w\lib\freexl_i.lib C:\OSGeo4w\lib\iconv.lib \
-		C:\OSGeo4W\lib\sqlite3_i.lib C:\OSGeo4W\lib\zlib.lib \
-		C:\OSGeo4W\lib\libxml2.lib C:\OSGeo4W\lib\librttopo.lib
+		$(LIBRARY_LIB)\proj.lib $(LIBRARY_LIB)\geos_c.lib \
+		$(LIBRARY_LIB)\freexl_i.lib $(LIBRARY_LIB)\iconv.lib \
+		$(LIBRARY_LIB)\sqlite3.lib $(LIBRARY_LIB)\zlib.lib \
+		$(LIBRARY_LIB)\libxml2.lib $(LIBRARY_LIB)\librttopo.lib \
+		$(LIBRARY_LIB)\charset.lib
 	if exist $(SPATIALITE_DLL).manifest mt -manifest \
 		$(SPATIALITE_DLL).manifest -outputresource:$(SPATIALITE_DLL);2
 		
diff -ru libspatialite-5.1.0-orig/nmake64.opt libspatialite-5.1.0/nmake64.opt
--- libspatialite-5.1.0-orig/nmake64.opt	2023-11-06 08:42:52.257245934 +1000
+++ libspatialite-5.1.0/nmake64.opt	2023-11-06 08:44:11.402686725 +1000
@@ -3,7 +3,7 @@
 
 # Uncomment the first for an optimized build, or the second for debug.
 OPTFLAGS=	/nologo /Ox /fp:precise /W4 /MD /D_CRT_SECURE_NO_WARNINGS \
-		/DDLL_EXPORT /DYY_NO_UNISTD_H
+		/DDLL_EXPORT /DYY_NO_UNISTD_H $(CF_FLAGS)
 #OPTFLAGS=	/nologo /Zi /MD /Fdspatialite.pdb /DDLL_EXPORT
 
 # Set the version number for the DLL.  Normally we leave this blank since
