diff -urN libspatialite-5.1.0-orig/test/Makefile.am libspatialite-5.1.0/test/Makefile.am
--- libspatialite-5.1.0-orig/test/Makefile.am	2023-09-17 11:28:28.531158194 -0500
+++ libspatialite-5.1.0/test/Makefile.am	2023-09-17 11:29:12.934750299 -0500
@@ -9,7 +9,6 @@
 		check_geometry_cols \
 		check_tempgeom \
 		check_create \
-		check_bufovflw \
 		check_fdo1 \
 		check_fdo2 \
 		check_fdo3 \
diff -urN libspatialite-5.1.0-orig/test/Makefile.in libspatialite-5.1.0/test/Makefile.in
--- libspatialite-5.1.0-orig/test/Makefile.in	2023-09-17 11:28:28.531158194 -0500
+++ libspatialite-5.1.0/test/Makefile.in	2023-09-17 11:29:08.528691547 -0500
@@ -91,7 +91,7 @@
 	check_init$(EXEEXT) check_init2$(EXEEXT) \
 	check_init_full$(EXEEXT) check_geom_aux$(EXEEXT) \
 	check_geometry_cols$(EXEEXT) check_tempgeom$(EXEEXT) \
-	check_create$(EXEEXT) check_bufovflw$(EXEEXT) \
+	check_create$(EXEEXT)  \
 	check_fdo1$(EXEEXT) check_fdo2$(EXEEXT) check_fdo3$(EXEEXT) \
 	check_fdo_bufovflw$(EXEEXT) check_md5$(EXEEXT) \
 	check_dbf_load$(EXEEXT) check_shp_load$(EXEEXT) \
@@ -217,9 +217,6 @@
 check_add_tile_triggers_bad_table_name_OBJECTS =  \
 	check_add_tile_triggers_bad_table_name.$(OBJEXT)
 check_add_tile_triggers_bad_table_name_LDADD = $(LDADD)
-check_bufovflw_SOURCES = check_bufovflw.c
-check_bufovflw_OBJECTS = check_bufovflw.$(OBJEXT)
-check_bufovflw_LDADD = $(LDADD)
 check_clone_table_SOURCES = check_clone_table.c
 check_clone_table_OBJECTS = check_clone_table.$(OBJEXT)
 check_clone_table_LDADD = $(LDADD)
@@ -587,7 +584,7 @@
 am__maybe_remake_depfiles = depfiles
 am__depfiles_remade = ./$(DEPDIR)/check_add_tile_triggers.Po \
 	./$(DEPDIR)/check_add_tile_triggers_bad_table_name.Po \
-	./$(DEPDIR)/check_bufovflw.Po ./$(DEPDIR)/check_clone_table.Po \
+	./$(DEPDIR)/check_clone_table.Po \
 	./$(DEPDIR)/check_control_points.Po \
 	./$(DEPDIR)/check_create.Po \
 	./$(DEPDIR)/check_createBaseTables.Po \
@@ -689,7 +686,7 @@
 am__v_CCLD_0 = @echo "  CCLD    " $@;
 am__v_CCLD_1 = 
 SOURCES = check_add_tile_triggers.c \
-	check_add_tile_triggers_bad_table_name.c check_bufovflw.c \
+	check_add_tile_triggers_bad_table_name.c \
 	check_clone_table.c check_control_points.c check_create.c \
 	check_createBaseTables.c check_cutter.c check_dbf_load.c \
 	check_drop_rename.c check_dxf.c check_endian.c check_exif.c \
@@ -736,7 +733,7 @@
 	shape_3d.c shape_cp1252.c shape_primitives.c shape_utf8_1.c \
 	shape_utf8_1ex.c shape_utf8_2.c
 DIST_SOURCES = check_add_tile_triggers.c \
-	check_add_tile_triggers_bad_table_name.c check_bufovflw.c \
+	check_add_tile_triggers_bad_table_name.c \
 	check_clone_table.c check_control_points.c check_create.c \
 	check_createBaseTables.c check_cutter.c check_dbf_load.c \
 	check_drop_rename.c check_dxf.c check_endian.c check_exif.c \
@@ -1371,10 +1368,6 @@
 	@rm -f check_add_tile_triggers_bad_table_name$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(check_add_tile_triggers_bad_table_name_OBJECTS) $(check_add_tile_triggers_bad_table_name_LDADD) $(LIBS)
 
-check_bufovflw$(EXEEXT): $(check_bufovflw_OBJECTS) $(check_bufovflw_DEPENDENCIES) $(EXTRA_check_bufovflw_DEPENDENCIES) 
-	@rm -f check_bufovflw$(EXEEXT)
-	$(AM_V_CCLD)$(LINK) $(check_bufovflw_OBJECTS) $(check_bufovflw_LDADD) $(LIBS)
-
 check_clone_table$(EXEEXT): $(check_clone_table_OBJECTS) $(check_clone_table_DEPENDENCIES) $(EXTRA_check_clone_table_DEPENDENCIES) 
 	@rm -f check_clone_table$(EXEEXT)
 	$(AM_V_CCLD)$(LINK) $(check_clone_table_OBJECTS) $(check_clone_table_LDADD) $(LIBS)
@@ -1803,7 +1796,6 @@
 
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/check_add_tile_triggers.Po@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/check_add_tile_triggers_bad_table_name.Po@am__quote@ # am--include-marker
-@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/check_bufovflw.Po@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/check_clone_table.Po@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/check_control_points.Po@am__quote@ # am--include-marker
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/check_create.Po@am__quote@ # am--include-marker
@@ -2246,13 +2238,6 @@
 	--log-file $$b.log --trs-file $$b.trs \
 	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
 	"$$tst" $(AM_TESTS_FD_REDIRECT)
-check_bufovflw.log: check_bufovflw$(EXEEXT)
-	@p='check_bufovflw$(EXEEXT)'; \
-	b='check_bufovflw'; \
-	$(am__check_pre) $(LOG_DRIVER) --test-name "$$f" \
-	--log-file $$b.log --trs-file $$b.trs \
-	$(am__common_driver_flags) $(AM_LOG_DRIVER_FLAGS) $(LOG_DRIVER_FLAGS) -- $(LOG_COMPILE) \
-	"$$tst" $(AM_TESTS_FD_REDIRECT)
 check_fdo1.log: check_fdo1$(EXEEXT)
 	@p='check_fdo1$(EXEEXT)'; \
 	b='check_fdo1'; \
@@ -3061,7 +3046,6 @@
 distclean: distclean-recursive
 		-rm -f ./$(DEPDIR)/check_add_tile_triggers.Po
 	-rm -f ./$(DEPDIR)/check_add_tile_triggers_bad_table_name.Po
-	-rm -f ./$(DEPDIR)/check_bufovflw.Po
 	-rm -f ./$(DEPDIR)/check_clone_table.Po
 	-rm -f ./$(DEPDIR)/check_control_points.Po
 	-rm -f ./$(DEPDIR)/check_create.Po
@@ -3214,7 +3198,6 @@
 maintainer-clean: maintainer-clean-recursive
 		-rm -f ./$(DEPDIR)/check_add_tile_triggers.Po
 	-rm -f ./$(DEPDIR)/check_add_tile_triggers_bad_table_name.Po
-	-rm -f ./$(DEPDIR)/check_bufovflw.Po
 	-rm -f ./$(DEPDIR)/check_clone_table.Po
 	-rm -f ./$(DEPDIR)/check_control_points.Po
 	-rm -f ./$(DEPDIR)/check_create.Po
